Toshi/Source/TObject.cpp

```diff
@@ -1,35 +1,35 @@
 #include "TObject.h"
 
-void __thiscall Toshi::TObject::Delete()
-{
-	if (this != 0)
-	{
-		this->DestructTObject(1);
-	}
-}
+// commented this because it's just a virtual constructor that is being created automatically
+// see this for more info:
+// https://stackoverflow.com/questions/8052146/whats-this-extra-parameter-passed-into-virtual-destructor
 
-void Toshi::TObject::DestructTObject(uint32_t param_2)
-{
-	if ((param_2 & 2) == 0)
-	{
-		Toshi::TObject::~TObject();
-		if ((param_2 & 1) != 0)
-		{
-			//tfree
-		}
-	}
-	else
-	{
-		
-	}
-}
+//void __thiscall Toshi::TObject::Delete()
+//{
+//	if (this != 0)
+//	{
+//		this->DestructTObject(1);
+//	}
+//}
 
-Toshi::TObject::~TObject()
-{
+// commented this because it's just a virtual constructor that is being created automatically
```

Comment:
I don't really understand. Is that tfree stuff in reality in the destructor or what?

Reply:
the third function pointer in the vtables of TObjects is a virtual destructor. any virtual destructor has one extra argument which controls if the data should be deleted at the end so you don't need to make those functions manually, just provide a destructor for a class that will tfree any dynamically allocated memory